server/tests/test_extraction.py

```diff
@@ -0,0 +1,297 @@
+"""Tests vision/extraction.py."""
+
+from typing import Any, Callable, Optional
+import numpy as np
+import pytest
+from unittest.mock import Mock
+
+from tabby_server.vision.extraction import (
+    ExtractionOption,
+    ExtractionResult,
+    extract_from_recognized_texts,
+    extract_option,
+    extract_result,
+)
+from tabby_server.vision.ocr import RecognizedText
+
+
+@pytest.fixture(scope="function")
+def mock_chat_completion() -> Callable[[Any], None]:
+    import openai.resources.chat
+
+    result = None
+
+    def mock_create(self, **kwargs) -> Any:
+        return result
+
+    openai.resources.chat.Completions.create = mock_create  # type: ignore
+
+    def set_result(new_result: Any) -> None:
+        nonlocal result
+        result = new_result
+
+    return set_result
+
+
+def test_extract_from_recognized_texts(mock_chat_completion):
+
+    array = np.array
+    int32 = np.int32
+    recognized_texts = [
+        RecognizedText(
+            text="Yhe",
+            corners=array(
+                [[1287, 1149], [1430, 1149], [1430, 1241], [1287, 1241]],
```

Comment:
How are these numbers generated? 
Do they represent what boxes the model should look at to find characters?